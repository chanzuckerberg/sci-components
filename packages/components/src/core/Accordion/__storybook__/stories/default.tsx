import { Args } from "@storybook/react";
import RawAccordion, {
  AccordionDetails,
  AccordionHeader,
} from "src/core/Accordion";
import { EXTRA_SHORT_LOREM_IPSUM } from "src/common/storybook/loremIpsum";

export const Accordion = (props: Args): JSX.Element => {
  const { id, subtitle, useDivider, togglePosition } = props;
  return (
    <RawAccordion
      id={id}
      useDivider={useDivider}
      togglePosition={togglePosition}
      {...props}
    >
      <AccordionHeader id={`${id}-header`} subtitle={subtitle}>
        Accordion Header
      </AccordionHeader>
      <AccordionDetails>{EXTRA_SHORT_LOREM_IPSUM}</AccordionDetails>
    </RawAccordion>
  );
};
