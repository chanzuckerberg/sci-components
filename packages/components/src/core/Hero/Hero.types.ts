import { ReactNode } from "react";

export interface HeroProps {
  /**
   * The main header text displayed prominently in the hero section
   */
  headerText: ReactNode;
  /**
   * Optional caption text that appears below the header
   */
  captionText?: ReactNode;
  /**
   * Optional content area that can contain any React node
   * This appears as a placeholder/content area in the design
   */
  children?: ReactNode;
  /**
   * Controls the maximum width of the hero container
   * @default "large"
   */
  maxWidth?: "small" | "medium" | "large";
  /**
   * Additional CSS class name for custom styling
   */
  className?: string;
  /**
   * Custom margin behavior - can override default container margin
   * Set to false to remove default margins
   * @default true
   */
  useContainerMargin?: boolean;
  /**
   * Background fill that can be either a color string or any React node (image, CardMedia, etc.)
   * When an image is provided, it will stretch to cover the entire component background
   */
  backgroundFill?: ReactNode;
  /**
   * The height of the hero container
   */
  height?: string;
  /**
   * Test ID for testing purposes
   */
  "data-testid"?: string;
}

// Breakpoint constants based on the design specifications
export const HERO_BREAKPOINTS = {
  SMALL: 768, // Mobile breakpoint
  MEDIUM: 1024, // Tablet breakpoint
  LARGE: 1200, // Desktop breakpoint
} as const;

// Default margin values based on the images
export const HERO_MARGINS = {
  SMALL: 24, // 24px margins on small screens
  MEDIUM: 40, // 40px margins on medium screens
  LARGE: 120, // 120px margins on large screens
} as const;
