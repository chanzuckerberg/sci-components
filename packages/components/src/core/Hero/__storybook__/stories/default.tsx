import { Args } from "@storybook/react-webpack5";
import RawHero from "src/core/Hero";

export const Hero = (props: Args): JSX.Element => {
  const { headerText, captionText, ...restProps } = props;

  return (
    <RawHero headerText={headerText} captionText={captionText} {...restProps} />
  );
};
