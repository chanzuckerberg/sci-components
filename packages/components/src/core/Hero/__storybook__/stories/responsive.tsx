import { Args } from "@storybook/react-webpack5";
import { Box } from "@mui/material";
import RawHero from "src/core/Hero";
import Button from "src/core/Button";

export const ResponsiveDemo = (props: Args): JSX.Element => {
  return (
    <Box sx={{ display: "flex", flexDirection: "column", gap: 4 }}>
      {/* Large Hero */}
      <Box>
        <h3>Large Maximum Width</h3>
        <RawHero
          headerText="Large Hero Section"
          captionText="This hero uses the large maximum width setting which provides more space for content on wider screens."
          maxWidth="large"
          {...props}
        >
          <Button sdsType="primary" sdsStyle="rounded">
            Get Started
          </Button>
        </RawHero>
      </Box>

      {/* Medium Hero */}
      <Box>
        <h3>Medium Maximum Width</h3>
        <RawHero
          headerText="Medium Hero Section"
          captionText="This hero uses a medium maximum width for more constrained layouts."
          maxWidth="medium"
          {...props}
        >
          <Button sdsType="secondary" sdsStyle="rounded">
            Learn More
          </Button>
        </RawHero>
      </Box>

      {/* Small Hero */}
      <Box>
        <h3>Small Maximum Width</h3>
        <RawHero
          headerText="Compact Hero"
          captionText="This hero is configured with a small maximum width for very focused layouts."
          maxWidth="small"
          {...props}
        >
          <Button sdsType="primary" sdsStyle="minimal">
            Explore
          </Button>
        </RawHero>
      </Box>
    </Box>
  );
};
