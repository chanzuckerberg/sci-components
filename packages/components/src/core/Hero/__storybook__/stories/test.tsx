import { Args } from "@storybook/react-webpack5";
import RawHero from "src/core/Hero";

export const TestDemo = (props: Args): JSX.Element => {
  return (
    <RawHero
      headerText="Test Hero Component"
      captionText="This is a test instance of the Hero component for snapshot testing and development."
      data-testid="hero"
      {...props}
    />
  );
};
