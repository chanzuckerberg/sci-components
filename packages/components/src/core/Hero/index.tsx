import React from "react";
import { CardMedia, CardMediaProps } from "@mui/material";
import {
  StyledHeroContainer,
  StyledContentWrapper,
  StyledHeaderText,
  StyledCaptionText,
  StyledContentArea,
  StyledPlaceholder,
  StyledBackgroundWrapper,
} from "./style";
import { HeroProps } from "./Hero.types";

export type { HeroProps };

const Hero = React.forwardRef<HTMLElement, HeroProps>((props, ref) => {
  const {
    headerText,
    captionText,
    children,
    maxWidth = "large",
    className,
    useContainerMargin = true,
    "data-testid": dataTestId = "hero",
    height,
    backgroundFill,
    ...restProps
  } = props;

  /**
   * Handle different types of backgroundFill similar to ContentCard image handling
   * - If it's a string, treat it as a color or URL
   * - If it's a CardMedia component, clone it and apply hero-specific styles
   * - If it's any other valid React element, clone it and apply styles
   * - If it's null/undefined, return null
   */
  const renderBackgroundFill = () => {
    if (!backgroundFill) return null;

    if (typeof backgroundFill === "string") {
      // If it's a string, check if it's a URL or color
      const isUrl =
        backgroundFill.startsWith("http") ||
        backgroundFill.startsWith("/") ||
        backgroundFill.startsWith("./");

      if (isUrl) {
        return (
          <CardMedia
            component="img"
            src={backgroundFill}
            alt="Hero Background"
            sx={{
              position: "absolute",
              top: 0,
              left: 0,
              width: "100%",
              height: "100%",
              objectFit: "cover",
              objectPosition: "center",
              zIndex: -1,
            }}
          />
        );
      } else {
        // It's a color string, handled by CSS
        return null;
      }
    } else if (
      React.isValidElement(backgroundFill) &&
      backgroundFill.type === CardMedia
    ) {
      // If it's already a CardMedia component, clone it and apply hero-specific styles
      return React.cloneElement(
        backgroundFill as React.ReactElement<CardMediaProps>,
        {
          sx: {
            ...(backgroundFill.props as CardMediaProps).sx,
            position: "absolute",
            top: 0,
            left: 0,
            width: "100%",
            height: "100%",
            objectFit: "cover",
            objectPosition: "center",
            zIndex: -1,
          },
        }
      );
    } else if (React.isValidElement(backgroundFill)) {
      // If it's any other valid React element, clone it and apply hero-specific styles
      return React.cloneElement(backgroundFill as React.ReactElement, {
        style: {
          position: "absolute",
          top: 0,
          left: 0,
          width: "100%",
          height: "100%",
          objectFit: "cover",
          objectPosition: "center",
          zIndex: -1,
          ...(backgroundFill.props as { style?: React.CSSProperties }).style,
        },
      });
    }

    return null;
  };

  return (
    <StyledHeroContainer
      ref={ref}
      className={className}
      maxWidth={maxWidth}
      useContainerMargin={useContainerMargin}
      data-testid={dataTestId}
      height={height}
      backgroundFill={backgroundFill}
      {...restProps}
    >
      <StyledBackgroundWrapper>
        {renderBackgroundFill()}
      </StyledBackgroundWrapper>
      <StyledContentWrapper maxWidth={maxWidth}>
        {headerText && <StyledHeaderText>{headerText}</StyledHeaderText>}

        {captionText && <StyledCaptionText>{captionText}</StyledCaptionText>}

        {children ? (
          <StyledContentArea>{children}</StyledContentArea>
        ) : (
          <StyledContentArea>
            <StyledPlaceholder>Content area (placeholder)</StyledPlaceholder>
          </StyledContentArea>
        )}
      </StyledContentWrapper>
    </StyledHeroContainer>
  );
});

Hero.displayName = "Hero";

export default Hero;
