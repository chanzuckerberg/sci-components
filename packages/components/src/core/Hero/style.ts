import styled from "@emotion/styled";
import { css } from "@emotion/react";
import {
  CommonThemeProps,
  fontTitleBoldL,
  fontTitleBoldM,
  fontTitleBoldS,
  fontBodyM,
  fontBodyS,
  getSemanticColors,
  getSpaces,
} from "../styles";
import { HeroProps, HERO_MARGINS } from "./Hero.types";

const doNotForwardProps = [
  "headerText",
  "captionText",
  "maxWidth",
  "useContainerMargin",
  "backgroundFill",
];

interface StyledHeroProps
  extends Omit<HeroProps, "headerText">,
    CommonThemeProps {
  headerText?: HeroProps["headerText"];
  backgroundFill?: HeroProps["backgroundFill"];
}

// Helper function to get responsive margins
const getResponsiveMargins = (useContainerMargin: boolean = true) => {
  if (!useContainerMargin)
    return css`
      margin: 0;
    `;

  return css`
    margin: 0 ${HERO_MARGINS.SMALL}px;

    @media (min-width: 768px) {
      margin: 0 ${HERO_MARGINS.MEDIUM}px;
    }

    @media (min-width: 1200px) {
      margin: 0 ${HERO_MARGINS.LARGE}px;
    }
  `;
};

// Helper function to get background styles based on backgroundFill prop
const getBackgroundStyles = (backgroundFill?: HeroProps["backgroundFill"]) => {
  if (!backgroundFill) return css``;

  if (typeof backgroundFill === "string") {
    // Check if it's a URL or color
    const isUrl =
      backgroundFill.startsWith("http") ||
      backgroundFill.startsWith("/") ||
      backgroundFill.startsWith("./");

    if (isUrl) {
      // If it's a URL, it will be handled by the CardMedia component
      return css``;
    } else {
      // If it's a string, treat it as a color
      return css`
        background-color: ${backgroundFill};
      `;
    }
  } else {
    // If it's a ReactNode, it will be handled by the component
    return css``;
  }
};

// Background wrapper for positioning background elements
export const StyledBackgroundWrapper = styled("div")`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
`;

// Main Hero Container
export const StyledHeroContainer = styled("section", {
  shouldForwardProp: (prop: string) => !doNotForwardProps.includes(prop),
})`
  ${(props: StyledHeroProps) => {
    const semanticColors = getSemanticColors(props);
    const spaces = getSpaces(props);

    const { backgroundFill, height } = props;

    return css`
      position: relative;
      width: 100%;
      background-color: ${backgroundFill
        ? "transparent"
        : semanticColors?.base?.backgroundPrimary};
      padding: ${spaces?.xl}px 0;
      height: ${height};
      ${getResponsiveMargins(props.useContainerMargin)}
      ${getBackgroundStyles(backgroundFill)}

      display: flex;
      align-items: center;
      text-align: center;

      @media (min-width: 768px) {
        padding: ${spaces?.xxl}px 0;
      }
    `;
  }}
`;

// Content Wrapper with max-width control
export const StyledContentWrapper = styled("div", {
  shouldForwardProp: (prop: string) => !doNotForwardProps.includes(prop),
})`
  ${(props: StyledHeroProps) => {
    const { maxWidth = "large" } = props;

    const maxWidthMap = {
      small: "600px",
      medium: "800px",
      large: "1000px",
    };

    return css`
      width: 100%;
      max-width: ${maxWidthMap[maxWidth]};
      display: flex;
      flex-direction: column;
      align-items: center;
      gap: ${props.theme?.spacing?.(3) || "24px"};
    `;
  }}
`;

// Header Text with responsive typography
export const StyledHeaderText = styled("h1", {
  shouldForwardProp: (prop: string) => !doNotForwardProps.includes(prop),
})`
  ${(props: CommonThemeProps) => {
    const semanticColors = getSemanticColors(props);

    return css`
      ${fontTitleBoldS(props)} // Small screens (26px)
      color: ${semanticColors?.base?.textPrimary};
      margin: 0;
      text-align: center;

      @media (min-width: 768px) {
        ${fontTitleBoldM(props)}// Medium screens (32px)
      }

      @media (min-width: 1200px) {
        ${fontTitleBoldL(props)}// Large screens (52px on wide, 40px on narrow)
      }
    `;
  }}
`;

// Caption Text with responsive typography
export const StyledCaptionText = styled("p", {
  shouldForwardProp: (prop: string) => !doNotForwardProps.includes(prop),
})`
  ${(props: CommonThemeProps) => {
    const semanticColors = getSemanticColors(props);
    const spaces = getSpaces(props);

    return css`
      ${fontBodyS(props)} // Small screens (14px)
      color: ${semanticColors?.base?.textSecondary};
      margin: 0;
      margin-top: ${spaces?.s}px;
      text-align: center;
      max-width: 80%;

      @media (min-width: 768px) {
        ${fontBodyM(props)} // Medium+ screens (16px on wide, 14px on narrow)
        max-width: 70%;
      }

      @media (min-width: 1200px) {
        max-width: 60%;
      }
    `;
  }}
`;

// Content Area for additional children
export const StyledContentArea = styled("div", {
  shouldForwardProp: (prop: string) => !doNotForwardProps.includes(prop),
})`
  ${(props: CommonThemeProps) => {
    const spaces = getSpaces(props);

    return css`
      width: 100%;
      display: flex;
      flex-direction: column;
      align-items: center;
      margin-top: ${spaces?.l}px;

      @media (min-width: 768px) {
        margin-top: ${spaces?.xl}px;
      }
    `;
  }}
`;

// Placeholder content box (for demonstration purposes)
export const StyledPlaceholder = styled("div", {
  shouldForwardProp: (prop: string) => !doNotForwardProps.includes(prop),
})`
  ${(props: CommonThemeProps) => {
    const semanticColors = getSemanticColors(props);
    const spaces = getSpaces(props);

    return css`
      width: 200px;
      height: 48px;
      background-color: ${semanticColors?.base?.backgroundSecondary};
      border: 2px dashed ${semanticColors?.base?.divider};
      border-radius: ${spaces?.xxxs}px;
      display: flex;
      align-items: center;
      justify-content: center;
      color: ${semanticColors?.base?.textSecondary};
      font-size: 12px;

      @media (min-width: 768px) {
        width: 280px;
        height: 56px;
        font-size: 14px;
      }

      @media (min-width: 1200px) {
        width: 320px;
        height: 64px;
      }
    `;
  }}
`;
