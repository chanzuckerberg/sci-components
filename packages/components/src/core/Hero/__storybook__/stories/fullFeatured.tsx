import { Args } from "@storybook/react-webpack5";
import RawHero from "src/core/Hero";
import { DemoContentSlot } from "../../style";

export const FullFeaturedDemo = (props: Args): JSX.Element => {
  const { headerText, captionText, ...restProps } = props;

  return (
    <RawHero headerText={headerText} captionText={captionText} {...restProps}>
      <DemoContentSlot>Content Slot (childNode)</DemoContentSlot>
    </RawHero>
  );
};
