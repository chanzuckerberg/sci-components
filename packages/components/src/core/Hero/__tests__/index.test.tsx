import { generateSnapshots } from "@chanzuckerberg/story-utils";
import { composeStories } from "@storybook/react-webpack5";
import { render, screen } from "@testing-library/react";
import React from "react";
import * as stories from "../__storybook__/index.stories";
import Hero from "../index";

// Returns a component that already contain all decorators from story level, meta level and global level.
const { Test } = composeStories(stories);

describe("<Hero />", () => {
  generateSnapshots(stories);

  it("renders hero component with basic props", () => {
    render(
      <Hero
        headerText="Test Header"
        captionText="Test Caption"
        data-testid="hero"
      />
    );

    const heroElement = screen.getByTestId("hero");
    expect(heroElement).not.toBeNull();

    const headerElement = screen.getByText("Test Header");
    expect(headerElement).toBeInTheDocument();

    const captionElement = screen.getByText("Test Caption");
    expect(captionElement).toBeInTheDocument();
  });

  it("renders without caption text", () => {
    render(<Hero headerText="Header Only" data-testid="hero" />);

    const heroElement = screen.getByTestId("hero");
    expect(heroElement).toBeInTheDocument();

    const headerElement = screen.getByText("Header Only");
    expect(headerElement).toBeInTheDocument();

    // Should show placeholder when no caption or children
    const placeholderElement = screen.getByText("Content area (placeholder)");
    expect(placeholderElement).toBeInTheDocument();
  });

  it("renders with custom children instead of placeholder", () => {
    render(
      <Hero headerText="Header with Custom Content" data-testid="hero">
        <button>Custom Button</button>
      </Hero>
    );

    const heroElement = screen.getByTestId("hero");
    expect(heroElement).toBeInTheDocument();

    const customButton = screen.getByText("Custom Button");
    expect(customButton).toBeInTheDocument();

    // Should not show placeholder when children are provided
    const placeholderElement = screen.queryByText("Content area (placeholder)");
    expect(placeholderElement).not.toBeInTheDocument();
  });

  it("applies maxWidth prop correctly", () => {
    const { rerender } = render(
      <Hero headerText="Header" maxWidth="small" data-testid="hero" />
    );

    let heroElement = screen.getByTestId("hero");
    expect(heroElement).toBeInTheDocument();

    // Test different maxWidth values
    rerender(<Hero headerText="Header" maxWidth="medium" data-testid="hero" />);

    heroElement = screen.getByTestId("hero");
    expect(heroElement).toBeInTheDocument();

    rerender(<Hero headerText="Header" maxWidth="large" data-testid="hero" />);

    heroElement = screen.getByTestId("hero");
    expect(heroElement).toBeInTheDocument();
  });

  it("handles useContainerMargin prop", () => {
    render(
      <Hero headerText="Header" useContainerMargin={false} data-testid="hero" />
    );

    const heroElement = screen.getByTestId("hero");
    expect(heroElement).toBeInTheDocument();
  });

  it("applies custom className", () => {
    render(
      <Hero
        headerText="Header"
        className="custom-hero-class"
        data-testid="hero"
      />
    );

    const heroElement = screen.getByTestId("hero");
    expect(heroElement).toHaveClass("custom-hero-class");
  });

  it("forwards ref correctly", () => {
    const ref = React.createRef<HTMLElement>();

    render(<Hero ref={ref} headerText="Header" data-testid="hero" />);

    expect(ref.current).toBeInstanceOf(HTMLElement);
    expect(ref.current?.tagName).toBe("SECTION");
  });

  it("renders with default data-testid when not provided", () => {
    render(<Hero headerText="Header" />);

    const heroElement = screen.getByTestId("hero");
    expect(heroElement).toBeInTheDocument();
  });

  it("renders JSX elements as headerText and captionText", () => {
    const headerElement = <span>JSX Header</span>;
    const captionElement = <span>JSX Caption</span>;

    render(
      <Hero
        headerText={headerElement}
        captionText={captionElement}
        data-testid="hero"
      />
    );

    expect(screen.getByText("JSX Header")).toBeInTheDocument();
    expect(screen.getByText("JSX Caption")).toBeInTheDocument();
  });

  it("renders with Test story configuration", () => {
    render(<Test {...Test.args} />);

    const heroElement = screen.getByTestId("hero");
    expect(heroElement).toBeInTheDocument();

    const headerElement = screen.getByText("Test Hero Component");
    expect(headerElement).toBeInTheDocument();
  });

  it("applies backgroundFill with color string", () => {
    render(
      <Hero
        headerText="Header with Color Background"
        backgroundFill="#ff0000"
        data-testid="hero"
      />
    );

    const heroElement = screen.getByTestId("hero");
    expect(heroElement).toBeInTheDocument();
    expect(heroElement).toHaveStyle("background-color: #ff0000");
  });

  it("applies backgroundFill with image object", () => {
    const imageBackground = {
      src: "https://example.com/image.jpg",
      alt: "Background image",
    };

    render(
      <Hero
        headerText="Header with Image Background"
        backgroundFill={imageBackground}
        data-testid="hero"
      />
    );

    const heroElement = screen.getByTestId("hero");
    expect(heroElement).toBeInTheDocument();
    expect(heroElement).toHaveStyle(
      "background-image: url(https://example.com/image.jpg)"
    );
    expect(heroElement).toHaveStyle("background-size: cover");
    expect(heroElement).toHaveStyle("background-position: center");
    expect(heroElement).toHaveStyle("background-repeat: no-repeat");
  });

  it("uses default background when backgroundFill is not provided", () => {
    render(
      <Hero headerText="Header with Default Background" data-testid="hero" />
    );

    const heroElement = screen.getByTestId("hero");
    expect(heroElement).toBeInTheDocument();
    // Should use the default semantic color background
    expect(heroElement).toBeInTheDocument();
  });
});
