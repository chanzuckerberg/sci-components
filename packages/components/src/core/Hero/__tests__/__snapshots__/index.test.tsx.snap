// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> BackgroundFillColor story renders snapshot 1`] = `
<section
  class="css-dh5ges"
  data-testid="hero"
  height="400px"
>
  <div
    class="css-7dmny"
  >
    <h1
      class="css-dkfd5g"
    >
      Hero with Color Background
    </h1>
    <p
      class="css-wzifq3"
    >
      This hero uses a color background fill instead of the default background color.
    </p>
    <div
      class="css-133yqvh"
    >
      <div
        class="css-tqgscj"
      >
        Content area (placeholder)
      </div>
    </div>
  </div>
</section>
`;

exports[`<Hero /> BackgroundFillImage story renders snapshot 1`] = `
<section
  class="css-1r31xwp"
  data-testid="hero"
  height="500px"
>
  <div
    class="css-7dmny"
  >
    <h1
      class="css-dkfd5g"
    >
      Hero with Image Background
    </h1>
    <p
      class="css-wzifq3"
    >
      This hero uses an image background that stretches to cover the entire component.
    </p>
    <div
      class="css-133yqvh"
    >
      <div
        class="css-tqgscj"
      >
        Content area (placeholder)
      </div>
    </div>
  </div>
</section>
`;

exports[`<Hero /> Default story renders snapshot 1`] = `
<section
  class="css-1pjlwce"
  data-testid="hero"
>
  <div
    class="css-7dmny"
  >
    <h1
      class="css-dkfd5g"
    >
      Header Text
    </h1>
    <p
      class="css-wzifq3"
    >
      Caption text Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna.
    </p>
    <div
      class="css-133yqvh"
    >
      <div
        class="css-tqgscj"
      >
        Content area (placeholder)
      </div>
    </div>
  </div>
</section>
`;

exports[`<Hero /> NoContainerMargins story renders snapshot 1`] = `
<section
  class="css-y8l1m5"
  data-testid="hero"
>
  <div
    class="css-7dmny"
  >
    <h1
      class="css-dkfd5g"
    >
      Full Width Hero
    </h1>
    <p
      class="css-wzifq3"
    >
      This hero extends to the full width without container margins.
    </p>
    <div
      class="css-133yqvh"
    >
      <div
        class="css-tqgscj"
      >
        Content area (placeholder)
      </div>
    </div>
  </div>
</section>
`;

exports[`<Hero /> SmallWidth story renders snapshot 1`] = `
<section
  class="css-1pjlwce"
  data-testid="hero"
>
  <div
    class="css-1ppzpm2"
  >
    <h1
      class="css-dkfd5g"
    >
      Compact Hero
    </h1>
    <p
      class="css-wzifq3"
    >
      A more compact hero design for focused layouts.
    </p>
    <div
      class="css-133yqvh"
    >
      <div
        class="css-tqgscj"
      >
        Content area (placeholder)
      </div>
    </div>
  </div>
</section>
`;

exports[`<Hero /> WithoutCaption story renders snapshot 1`] = `
<section
  class="css-1pjlwce"
  data-testid="hero"
>
  <div
    class="css-7dmny"
  >
    <h1
      class="css-dkfd5g"
    >
      Header Text Only
    </h1>
    <div
      class="css-133yqvh"
    >
      <div
        class="css-tqgscj"
      >
        Content area (placeholder)
      </div>
    </div>
  </div>
</section>
`;
